### PR TITLE
[EXPORTER] Fix the format of SpanLink for ETW

### PR DESCRIPTION
**Pull request by ThomsonTan**
## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h [hunk 1]
```diff
@@ -214,17 +214,38 @@ class Tracer : public opentelemetry::trace::Tracer,
     {
       size_t idx = 0;
       std::string linksValue;
+
+      // reserve space for all the SpanLinks.
+      // A single SpanLink will be outptut as:
+      // [{"toSpanId":"9a43c801557f26b7","toTraceId":"ac6cd70ac4bb168a99cb7651b048d965"}]
+      // The second and above link output to string are 1 byte less than the first link.
+      const size_t linkValueSize = 80;
```
Comment (lalitb) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h:222]: the max allowed link size is 128, as per the specs - https://github.com/open-telemetry/opentelemetry-specification/blob/625e0f8500f2aea48982a943997c991ff41ee9bb/specification/trace/sdk.md#span-limits

File: exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h [hunk 2]
```diff
@@ -214,17 +214,38 @@ class Tracer : public opentelemetry::trace::Tracer,
...
+      linksValue.reserve(linkValueSize + (links.size() - 1) * (linkValueSize - 1));
+      linksValue += "[";
+
       links.ForEachKeyValue([&](opentelemetry::trace::SpanContext ctx,
                                 const opentelemetry::common::KeyValueIterable &) {
-        if (!linksValue.empty())
+        if (idx == 0)
```
Comment (lalitb) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h:228]: nit - we can have a bool for this ?

File: exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h (hunk 1 above)
Comment (lalitb) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h:222]: nit -where this 80 is coming from, perhaps mention in comment? Also naming guideline - https://google.github.io/styleguide/cppguide.html#Constant_Names

Comment (ThomsonTan) [reply exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h]: Changed the name, and the new name also documents the purpose of this variable.

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for the PR

**Review by marcalff — APPROVED** [review]

**Pull request merged**